### PR TITLE
Add Capabilities flag to LSP 

### PR DESCRIPTION
**Pull request by MikeStall**
Add Capabilities flag to LSP. 
Add Fx2NL hook on NLHandler. 
Update NL2Fx messages .

### CODE REVIEW

File: src/libraries/Microsoft.PowerFx.LanguageServerProtocol/LanguageServer/LanguageServer.cs [hunk 1]
```diff
@@ -135,9 +135,17 @@ public void OnDataReceived(string jsonRpcPayload)
                         case CustomProtocolNames.CommandExecuted:
                             HandleCommandExecutedRequest(id, paramsJson);
                             break;
+
+                        case CustomProtocolNames.GetCapabilities:
```
Comment (akshar2401) [inline src/libraries/Microsoft.PowerFx.LanguageServerProtocol/LanguageServer/LanguageServer.cs:139]: Happened to come across this. Why do we need Custom capabilities protocol when language server has a well defined initialize protocol that includes capabilities? I understand the need we need to get this done quickly but we should follow the protocol whenever we can

Comment (akshar2401) [reply src/libraries/Microsoft.PowerFx.LanguageServerProtocol/LanguageServer/LanguageServer.cs:139]: https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#initialize

**Comment by LucGenetier** [comment]
Public API changes: <br />❌CP0002 M:Microsoft.PowerFx.LanguageServerProtocol.NLHandler.NL2Fx(Microsoft.PowerFx.LanguageServerProtocol.NL2FxParameters,System.Threading.CancellationToken)

File: src/libraries/Microsoft.PowerFx.LanguageServerProtocol/LanguageServer/LanguageServer.cs (hunk 1 above)
Comment (akshar2401) [reply src/libraries/Microsoft.PowerFx.LanguageServerProtocol/LanguageServer/LanguageServer.cs:139]: Chatted offline. Given the need of the hour, it should be okay to get this in as long as we follow up to do it the way protocol specifies.

Comment (MikeStall) [reply src/libraries/Microsoft.PowerFx.LanguageServerProtocol/LanguageServer/LanguageServer.cs:139]: Tracking with #1843

**Pull request merged**